toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -298,9 +298,13 @@ func (g *PkgGraph) validateNodeForLookup(pkgNode *PkgNode) (valid bool, err erro
 		switch pkgNode.Type {
 		case TypeLocalBuild:
 			haveDuplicateNode = existingLookup.BuildNode != nil
-		// For the purposes of lookup, a "Remote" node provides the same utility as a "Run" node
-		case TypeLocalRun, TypeRemoteRun:
-			haveDuplicateNode = existingLookup.RunNode != nil
+		case TypeLocalRun:
+			haveDuplicateNode = (existingLookup.RunNode != nil) && (existingLookup.RunNode.Type == TypeLocalRun)
+		case TypeRemoteRun:
+			// It is possible that
+			// 1. LocalRun and RemoteRun co-exist in a graph
+			// 2. Multiple RemoteRun nodes can exist in a graph
```

Comment:
Maybe we can leave for a future fix, but I'd like to see if we can fix this so there is only a single remote node for a given version requirement in the graph. (and do the same for the prebuilt nodes if they suffer the same issue). I don't think they will cause any issues with cycles since both prebuilt and remote nodes by definition have no dependencies, so everywhere we need one, we can use the same one.

Reply:
Yes. It's an issue. But I want to fix in a separate PR. I don't want to bloat this PR with multiple changes. Because upon regressions, it will be hard to identify the root cause.